app/services/progress_service.rb

```diff
@@ -0,0 +1,37 @@
+class ProgressService
+  attr_reader :status, :trainee
+
+  class << self
+    def call(**args)
+      new(**args).call
+    end
+  end
+
+  def initialize(validator:, progress_value:)
+    @validator = validator
+    @progress_value = progress_value
+  end
+
+  def call
+    return Progress::STATUSES[:in_progress] if in_progress?
+    return Progress::STATUSES[:completed] if completed?
+
+    Progress::STATUSES[:not_started]
+  end
+
+private
+
+  attr_reader :validator, :progress_value
+
+  def started?
+    @validator.fields.values.flatten.compact.any?
+  end
+
+  def in_progress?
+    started? && (!@validator.valid? || @validator.valid?) && !progress_value
```

Comment:
`(!@validator.valid? || @validator.valid?)` has no effect on the overall statment since it's basically saying "valid or invalid". If I remove it, all the tests continue to pass.

Reply:
This should really just check for `started? && !completed?`, have made the change in https://github.com/DFE-Digital/register-trainee-teachers/pull/161/commits/83d9093f98e0da85074cf5c4e302fc49ba9369d0